### PR TITLE
qt5-build.eclass: fix avoiding qmake recompilation for Qt 5.8 and later

### PR DESCRIPTION
**Pull request by kensington**
The old hack is no longer possible since qt/qtbase@4ce0beee1b69a8695fc24a244a8a3053711906ac. This is a much worse hack but I'm not sure what other option we have.

### CODE REVIEW

File: eclass/qt5-build.eclass [hunk 1]
```diff
@@ -183,8 +183,13 @@ qt5-build_src_prepare() {
 		qt5_symlink_tools_to_build_dir
 
 		# Avoid unnecessary qmake recompilations
-		sed -i -re "s|^if true;.*(\[ '\!').*(\"\\\$outpath/bin/qmake\".*)|if \1 -e \2 then|" \
-			configure || die "sed failed (skip qmake bootstrap)"
+		if [[ ${QT5_MINOR_VERSION} -ge 8 ]]; then
+			sed -e "/Creating qmake/i if false; then" -e '/cd "$outpath\/qmake"; "$MAKE" -s/a fi' \
```
Comment (Pesa) [inline eclass/qt5-build.eclass]: why not adding `fi` after `echo "Done."`?

also, keep one sed expression per line, so break before the second `-e`, and use single quotes for the first sed expression as well

File: eclass/qt5-build.eclass [hunk 2]
```diff
@@ -183,8 +183,13 @@ qt5-build_src_prepare() {
...
+				-i configure || die "sed failed (skip qmake bootstrap)"
```
Comment (Pesa) [inline eclass/qt5-build.eclass]: put the `-i` right after `sed` on the previous line, for consistency

**Comment by kensington** [comment]
Updated.

File: eclass/qt5-build.eclass [hunk 3]
```diff
@@ -179,8 +179,14 @@ qt5-build_src_prepare() {
 		qt5_symlink_tools_to_build_dir
 
 		# Avoid unnecessary qmake recompilations
-		sed -i -re "s|^if true;.*(\[ '\!').*(\"\\\$outpath/bin/qmake\".*)|if \1 -e \2 then|" \
-			configure || die "sed failed (skip qmake bootstrap)"
+		if [[ ${QT5_MINOR_VERSION} -ge 8 ]]; then
+			sed -i -e '/Creating qmake/i if false; then' \
```
Comment (Pesa) [inline eclass/qt5-build.eclass]: hmm this disables the build unconditionally though right? how does qmake get built in `qtcore`?

Comment (kensington) [inline eclass/qt5-build.eclass]: Forgive me, I completely missed the purpose of the original code.

File: eclass/qt5-build.eclass [hunk 4]
```diff
@@ -179,8 +179,14 @@ qt5-build_src_prepare() {
 		qt5_symlink_tools_to_build_dir
 
 		# Avoid unnecessary qmake recompilations
-		sed -i -re "s|^if true;.*(\[ '\!').*(\"\\\$outpath/bin/qmake\".*)|if \1 -e \2 then|" \
-			configure || die "sed failed (skip qmake bootstrap)"
+		if [[ ${QT5_MINOR_VERSION} -ge 8 ]]; then
+			sed -i -e '/Creating qmake/i if [ '!' -e "$outpath/bin/qmake" ]; then' \
```
Comment (Pesa) [inline eclass/qt5-build.eclass]: hmm there's a quoting problem here, I'm afraid you have to use double quotes as outermost quotes (for the whole sed expression) and escape them when used as literal quotes around `$outpath/bin/qmake`. Also don't forget to escape the `$`.

Comment (kensington) [inline eclass/qt5-build.eclass]: Indeed, I'll fix in a minute. Curiously, it does still work however!

**Comment by kensington** [comment]
Picked as 241a0712deb4e10edaa70f6681490a97d49d5126.

**Pull request closed without merging**